src/Aspire.Cli/Backchannel/AppHostBackchannel.cs

```diff
@@ -77,4 +81,30 @@ public async Task ConnectAsync(string socketPath, CancellationToken cancellation
 
         _rpcTaskCompletionSource.SetResult(rpc);
     }
+
+    public async Task<string[]> GetPublishersAsync(CancellationToken cancellationToken)
+    {
+        var rpc = await _rpcTaskCompletionSource.Task.ConfigureAwait(false);
+
+        logger.LogDebug("Requesting publishers");
+
+        var publishers = await rpc.InvokeWithCancellationAsync<string[]>(
+            "GetPublishersAsync",
+            Array.Empty<object>(),
+            cancellationToken).ConfigureAwait(false);
+
+        return publishers;
+    }
+
+    public void Dispose()
```

Comment:
What's this for?

Reply:
It was either this or making an RPC call to get the apphost to tear itself down. This seemed the simpler option.